lib/pure/memfiles.nim

```diff
@@ -144,14 +170,16 @@ proc open*(filename: string, mode: FileMode = fmRead,
 
     # since the strings are always 'nil', we simply always call
     # CreateFileMappingW which should be slightly faster anyway:
-    result.mapHandle = createFileMappingW(
+    let mapHandle = createFileMappingW(
       result.fHandle, nil,
       if readonly: PAGE_READONLY else: PAGE_READWRITE,
       0, 0, nil)
 
-    if result.mapHandle == 0:
+    if mapHandle == nil:
```

Comment:
How can this be correct? It used to check against 0, now it checks against 'nil'.